File: src/main/kotlin/lucasalfare/basicappengine/util/ElapsedTimeChecker.java
package lucasalfare.basicappengine.util;

public class ElapsedTimeChecker {

    public static void check(Action actionToCheck, int times, boolean useNanos) {
        long begin = useNanos ? System.nanoTime() : System.currentTimeMillis();
        for (int i = 0; i < times; i++) {
            actionToCheck.perform();
        }

        long end = (useNanos ? System.nanoTime() : System.currentTimeMillis()) - begin;
        System.out.println("Spent " + end + (useNanos ? "ns" : "ms") + " to perform " + times + " times.");
    }

    public interface Action {
        void perform();
    }
}

File: src/main/kotlin/lucasalfare/basicappengine/util/ElapsedTimeChecker.kt
package lucasalfare.basicappengine.util

object ElapsedTimeChecker {
  fun check(times: Int, useNanos: Boolean, action: () -> Unit) {
    val begin = if (useNanos) System.nanoTime() else System.currentTimeMillis()
    repeat(times) { action() }
    val end = (if (useNanos) System.nanoTime() else System.currentTimeMillis()) - begin
    println("Spent " + end + (if (useNanos) "ns" else "ms") + " to perform the passed action " + times + " times.")
  }
}